AdminWebsite/AdminWebsite/ClientApp/src/app/booking/participant/list/participant-list.component.ts

```diff
@@ -61,46 +61,64 @@ export class ParticipantListComponent implements OnInit, OnChanges, DoCheck {
         }
-        const judges = this.hearing.participants.filter(participant => participant.is_judge);
-        const staffMembers = this.hearing.participants.filter(
-            participant => participant.hearing_role_name === Constants.HearingRoles.StaffMember
-        );
-        const panelMembersAndWingers = this.hearing.participants.filter(participant =>
-            Constants.JudiciaryRoles.includes(participant.hearing_role_name)
-        );
-
-        const interpretersAndInterpretees = this.getInterpreterAndInterpretees();
-        const others = this.hearing.participants.filter(
-            participant =>
-                !participant.is_judge &&
-                !Constants.OtherParticipantRoles.includes(participant.hearing_role_name) &&
-                !interpretersAndInterpretees.includes(participant)
-        );
-        const observers = this.hearing.participants.filter(
-            participant =>
-                participant.hearing_role_name === Constants.HearingRoles.Observer && !interpretersAndInterpretees.includes(participant)
-        );
-
-        this.sortedParticipants = [
-            ...judges,
-            ...panelMembersAndWingers,
-            ...staffMembers,
-            ...others,
-            ...interpretersAndInterpretees,
-            ...observers
-        ];
+        const judges = this.hearing.participants.filter(participant => participant.is_judge).sort(compareByPartyThenByFirstName());
+
+        const staffMembers = this.hearing.participants
+            .filter(participant => participant.hearing_role_name === Constants.HearingRoles.StaffMember)
+            .sort(compareByPartyThenByFirstName());
+
+        const panelMembers = this.hearing.participants
+            .filter(participant =>
+                Constants.JudiciaryRoles.includes(
+                    participant.case_role_name === Constants.None ? participant.hearing_role_name : participant.case_role_name
+                )
+            )
+            .sort(compareByPartyThenByFirstName());
+
+        const observers = this.hearing.participants
+            .filter(
+                participant =>
+                    Constants.HearingRoles.Observer ===
+                    (participant.case_role_name === Constants.None ? participant.hearing_role_name : participant.case_role_name)
+            )
+            .sort(compareByPartyThenByFirstName());
+
+        const others = this.hearing.participants
+            .filter(
+                participant =>
+                    !participant.is_judge &&
+                    !staffMembers.includes(participant) &&
+                    !panelMembers.includes(participant) &&
+                    !observers.includes(participant) &&
+                    participant.hearing_role_name !== Constants.HearingRoles.Interpreter
+            )
+            .sort(compareByPartyThenByFirstName());
+
+        const sortedList = [...judges, ...panelMembers, ...staffMembers, ...others, ...observers];
```

Comment:
Is Interpreter and Interprettees not part of the roles now ?

Reply:
I've removed interpreteres/interpretees. The interpreters are inserted into the the sorted list underneath wherever their interpretee counterpart is, so they are always beneath them